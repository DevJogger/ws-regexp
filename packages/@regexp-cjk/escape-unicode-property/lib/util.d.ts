import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
export declare function handleOptions(options: IOptionsRewritePattern, flags?: string): {
    options: IOptionsRewritePattern;
    flags: string;
};
