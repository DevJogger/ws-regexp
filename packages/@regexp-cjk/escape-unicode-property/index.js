"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.escapeUnicodePropertyPattern = exports.escapeUnicodePropertyPatternCore = exports.replaceUnicodePropertyPattern = exports.matchUnicodePropertyPattern = exports.isUnicodePropertyPattern = exports.hasUnicodePropertyPattern = void 0;
const rewrite_pattern_1 = require("@regexp-cjk/rewrite-pattern");
const util_1 = require("./lib/util");
function hasUnicodePropertyPattern(source) {
    return /\\[pP]\{[A-Z][\w=_]*}/.test(source);
}
exports.hasUnicodePropertyPattern = hasUnicodePropertyPattern;
function isUnicodePropertyPattern(source) {
    return /^\\[pP]\{[A-Z][\w=_]*}$/.test(source);
}
exports.isUnicodePropertyPattern = isUnicodePropertyPattern;
function matchUnicodePropertyPattern(source) {
    return /\\([pP])\{([A-Z][\w_]*)(?:=([\w_]+))?}/.exec(source);
}
exports.matchUnicodePropertyPattern = matchUnicodePropertyPattern;
function replaceUnicodePropertyPattern(source, cb) {
    return source.replace(/\\([pP])\{([A-Z][\w_]*)(?:=([\w_]+))?}/g, cb);
}
exports.replaceUnicodePropertyPattern = replaceUnicodePropertyPattern;
function escapeUnicodePropertyPatternCore(source, flags, options) {
    return rewrite_pattern_1.rewritePatternCore(source, flags !== null && flags !== void 0 ? flags : 'u', {
        ...options,
        unicodePropertyEscape: true,
    });
}
exports.escapeUnicodePropertyPatternCore = escapeUnicodePropertyPatternCore;
function escapeUnicodePropertyPattern(source, flags, options) {
    ({ flags, options } = util_1.handleOptions(options, flags));
    return rewrite_pattern_1.rewritePatternCore(source, flags, options);
}
exports.escapeUnicodePropertyPattern = escapeUnicodePropertyPattern;
exports.default = escapeUnicodePropertyPattern;
//# sourceMappingURL=index.js.map