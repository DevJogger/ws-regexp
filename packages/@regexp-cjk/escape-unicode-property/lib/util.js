"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = void 0;
const util_1 = require("@regexp-cjk/rewrite-pattern/lib/util");
const lodash_1 = require("lodash");
function handleOptions(options, flags) {
    return (0, util_1.handleOptions)((0, lodash_1.merge)({}, options, {
        unicodePropertyEscape: true,
        rewriteFlags: {
            unicode: true,
        },
    }), flags !== null && flags !== void 0 ? flags : 'u');
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map