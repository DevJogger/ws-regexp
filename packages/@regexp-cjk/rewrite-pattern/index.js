"use strict";
/**
 * Created by user on 2020/6/8.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.rewritePatternCore = rewritePatternCore;
exports.rewritePattern = rewritePattern;
exports.rewritePatternByRegExpCore = rewritePatternByRegExpCore;
exports.rewritePatternByRegExp = rewritePatternByRegExp;
exports.rewriteRegExpCore = rewriteRegExpCore;
exports.rewriteRegExp = rewriteRegExp;
const util_1 = require("./lib/util");
const regexpu_core_v4_1 = require("@regexp-cjk/regexpu-core-v4");
function rewritePatternCore(source, flags, options) {
    return (0, regexpu_core_v4_1.rewritePatternV4)(source, flags, options);
}
function rewritePattern(source, flags, options) {
    ({ options, flags } = (0, util_1.handleOptions)(options, flags));
    return (0, regexpu_core_v4_1.rewritePatternV4)(source, flags, options);
}
function rewritePatternByRegExpCore(re, options) {
    var _a;
    return rewritePattern(re.source, (_a = options === null || options === void 0 ? void 0 : options.flags) !== null && _a !== void 0 ? _a : re.flags, options);
}
function rewritePatternByRegExp(re, options) {
    var _a;
    return rewritePattern(re.source, (_a = options === null || options === void 0 ? void 0 : options.flags) !== null && _a !== void 0 ? _a : re.flags, options);
}
function rewriteRegExpCore(re, options) {
    var _a;
    let flags = (_a = options === null || options === void 0 ? void 0 : options.flags) !== null && _a !== void 0 ? _a : re.flags;
    let source = (0, regexpu_core_v4_1.rewritePatternV4)(re.source, flags, options);
    return new RegExp(source, flags);
}
function rewriteRegExp(re, options) {
    let flags;
    ({ options, flags } = (0, util_1.handleOptions)(options, re.flags));
    let source = (0, regexpu_core_v4_1.rewritePatternV4)(re.source, flags, options);
    return new RegExp(source, flags);
}
exports.default = rewritePattern;
//# sourceMappingURL=index.js.map