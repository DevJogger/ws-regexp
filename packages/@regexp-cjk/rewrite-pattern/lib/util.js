"use strict";
/**
 * Created by user on 2020/6/8.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = exports.rewritePatternOptions = void 0;
const lodash_1 = require("lodash");
const rewrite_flags_1 = require("@regexp-cjk/rewrite-flags");
function rewritePatternOptions(options, flags) {
    var _a;
    return (0, lodash_1.defaults)({}, options, {
        dotAllFlag: true,
        lookbehind: true,
        namedGroup: true,
        useUnicodeFlag: (_a = flags === null || flags === void 0 ? void 0 : flags.includes) === null || _a === void 0 ? void 0 : _a.call(flags, 'u'),
    });
}
exports.rewritePatternOptions = rewritePatternOptions;
function handleOptions(options, flags) {
    var _a;
    flags = (_a = options === null || options === void 0 ? void 0 : options.flags) !== null && _a !== void 0 ? _a : flags;
    if (options === null || options === void 0 ? void 0 : options.rewriteFlags) {
        flags = (0, rewrite_flags_1.rewriteFlags)(flags, options.rewriteFlags);
    }
    options = rewritePatternOptions(options, flags);
    delete options.flags;
    return {
        options,
        flags,
    };
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map