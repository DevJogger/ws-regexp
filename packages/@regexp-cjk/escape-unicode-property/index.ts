import {
	rewritePattern,
	rewritePatternCore,
	IOptionsRewritePatternCore,
	IOptionsRewritePattern,
} from '@regexp-cjk/rewrite-pattern';
import { handleOptions } from './lib/util';

export function hasUnicodePropertyPattern(source: string)
{
	return /\\[pP]\{[A-Z][\w=_]*}/.test(source)
}

export function isUnicodePropertyPattern(source: string)
{
	return /^\\[pP]\{[A-Z][\w=_]*}$/.test(source)
}

export interface IUnicodePropertyRegExpExecArray extends RegExpExecArray
{
	/**
	 * \\p{UnicodePropertyName=UnicodePropertyValue}
	 */
	0: string,
	/**
	 * [pP]
	 */
	1: string,
	/**
	 * UnicodePropertyName
	 */
	2: string,
	/**
	 * UnicodePropertyValue
	 */
	3?: string,
}

export function matchUnicodePropertyPattern(source: string): IUnicodePropertyRegExpExecArray
{
	return /\\([pP])\{([A-Z][\w_]*)(?:=([\w_]+))?}/.exec(source) as any
}

export function replaceUnicodePropertyPattern(source: string,
	cb: (substring: string, p: string | 'P' | 'p', name: string, value?: string, ...any: any[]) => string,
)
{
	return source.replace(/\\([pP])\{([A-Z][\w_]*)(?:=([\w_]+))?}/g, cb as any)
}

export function escapeUnicodePropertyPatternCore(source: string, flags?: string, options?: IOptionsRewritePatternCore)
{
	return rewritePatternCore(source, flags ?? 'u', {
		...options,
		unicodePropertyEscape: true,
	})
}

export function escapeUnicodePropertyPattern(source: string, flags?: string, options?: IOptionsRewritePattern)
{
	({ flags, options } = handleOptions(options, flags));

	return rewritePatternCore(source, flags, options)
}

export default escapeUnicodePropertyPattern
