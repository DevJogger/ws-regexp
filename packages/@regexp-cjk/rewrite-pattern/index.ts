/**
 * Created by user on 2020/6/8.
 */

import _rewritePattern from 'regexpu-core';
import { rewriteFlags, IOptionsRewriteFlags } from '@regexp-cjk/rewrite-flags';
import { rewritePatternOptions, handleOptions } from './lib/util';

/**
 * The optional options argument recognizes the following properties:
 */
export interface IOptionsRewritePatternCore
{
	/**
	 * Setting this option to true enables support for the s (dotAll) flag.
	 */
	dotAllFlag?: boolean,

	/**
	 * Setting this option to true enables support for Unicode property escapes:
	 */
	unicodePropertyEscape?: boolean,

	/**
	 * Setting this option to true enables support for lookbehind assertions.
	 */
	lookbehind?: boolean,

	/**
	 * Setting this option to true enables support for named capture groups.
	 */
	namedGroup?: boolean,

	/**
	 * This option is a function that gets called when a named capture group is found.
	 * It receives two parameters:
	 * the name of the group, and its index.
	 */
	onNamedGroup?(name: string, index: number): void

	/**
	 * Setting this option to true enables the use of Unicode code point escapes of the form \u{â€¦}.
	 * Note that in regular expressions,
	 * such escape sequences only work correctly when the ES2015 u flag is set.
	 * Enabling this setting often results in more compact output,
	 * although there are cases (such as \p{Lu}) where it actually increases the output size.
	 */
	useUnicodeFlag?: boolean,
}

export interface IOptionsRewritePattern extends IOptionsRewritePatternCore
{
	flags?: string,

	rewriteFlags?: IOptionsRewriteFlags,
}

export function rewritePatternCore(source: string, flags?: string, options?: IOptionsRewritePatternCore): string
{
	return _rewritePattern(source, flags, options)
}

export function rewritePattern(source: string, flags?: string, options?: IOptionsRewritePattern): string
{
	({ options, flags } = handleOptions(options, flags));

	return _rewritePattern(source, flags, options)
}

export function rewritePatternByRegExpCore(re: RegExp, options?: IOptionsRewritePattern): string
{
	return rewritePattern(re.source, options?.flags ?? re.flags, options)
}

export function rewritePatternByRegExp(re: RegExp, options?: IOptionsRewritePattern): string
{
	return rewritePattern(re.source, options?.flags ?? re.flags, options)
}

export function rewriteRegExpCore(re: RegExp, options?: IOptionsRewritePattern)
{
	let flags = options?.flags ?? re.flags;

	let source = _rewritePattern(re.source, flags, options)

	return new RegExp(source, flags)
}

export function rewriteRegExp(re: RegExp, options?: IOptionsRewritePattern)
{
	let flags: string;

	({ options, flags } = handleOptions(options, re.flags));

	let source = _rewritePattern(re.source, flags, options)

	return new RegExp(source, flags)
}

export default rewritePattern
