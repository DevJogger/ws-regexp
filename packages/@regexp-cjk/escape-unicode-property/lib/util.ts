import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
import { handleOptions as _handleOptions } from '@regexp-cjk/rewrite-pattern/lib/util';
import { merge } from 'lodash';

export function handleOptions(options: IOptionsRewritePattern, flags?: string)
{
	return _handleOptions(merge({}, options, {
		unicodePropertyEscape: true,
		rewriteFlags: {
			unicode: true,
		},
	}), flags ?? 'u');
}
