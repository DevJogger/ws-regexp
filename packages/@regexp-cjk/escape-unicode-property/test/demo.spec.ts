import {
	escapeUnicodePropertyPattern,
	matchUnicodePropertyPattern,
	hasUnicodePropertyPattern,
	isUnicodePropertyPattern, replaceUnicodePropertyPattern,
} from '../index';
import { inspect } from 'util';
import escapeStringRegexp from 'escape-string-regexp';
import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
import { handleOptions } from '../lib/util';
import { EnumFlagMap } from '@regexp-cjk/rewrite-pattern/flags';

const knownUnicodeProperty = [
	`\\p{Emoji_Presentation}`,
	`\\P{Script_Extensions=Latin}`,
];

const unknownUnicodeProperty = [
	`\\p{UnicodePropertyName=UnicodePropertyValue}`,
	`\\P{UnicodeBinaryPropertyName}`,
];

describe(`escapeUnicodePropertyPattern`, () =>
{
	describe(`allow known property`, () =>
	{
		knownUnicodeProperty.forEach(source =>
		{
			test(inspect(source), () =>
			{
				expect(isUnicodePropertyPattern(source)).toStrictEqual(true);

				let actual = escapeUnicodePropertyPattern(source);
				let actual2 = matchUnicodePropertyPattern(source);

				let expected;

				//expect(actual).toStrictEqual(expected);
				//expect(actual).toBeInstanceOf(Date);
				expect(actual).toMatchSnapshot();
				expect(actual2).toMatchSnapshot();

			});
		})
	})

	describe(`throw error when unknown property`, () =>
	{
		unknownUnicodeProperty.forEach(source =>
		{
			test(inspect(source), () =>
			{

				expect(() => escapeUnicodePropertyPattern(source)).toThrowError()

			});
		})
	});

})

describe(`allow any property like`, () =>
{

	unknownUnicodeProperty.forEach(source =>
	{
		test(inspect(source), () =>
		{

			let actual = matchUnicodePropertyPattern(source);
			let actual2 = isUnicodePropertyPattern(source);
			let actual3 = hasUnicodePropertyPattern(source);

			expect(actual2).toStrictEqual(true);
			expect(actual3).toStrictEqual(true);

			expect(actual).toMatchSnapshot();

		});
	})

})

describe(`replaceUnicodePropertyPattern`, () =>
{

	knownUnicodeProperty
		.concat(unknownUnicodeProperty)
		.forEach(source =>
		{
			let input = `xxxx${source}yyyy`;

			test(inspect(input), () =>
			{

				let actual = replaceUnicodePropertyPattern(input, (substring, p, name, value, any) => {
					return ''
				});

				let actual2 = replaceUnicodePropertyPattern(input, (substring, p, name, value, any) => {
					return `\\p{P}`
				});

				expect(actual).toStrictEqual(`xxxxyyyy`);

				expect(actual2).toStrictEqual(`xxxx\\p{P}yyyy`);

				expect(hasUnicodePropertyPattern(actual2)).toBeTruthy();

			})
		})
	;

})

describe(`handleOptions`, () =>
{

	([
		{},
		{
			flags: "",
		},
		{
			options: {
				rewriteFlags: {
					unicode: false,
				}
			},
		},
		{
			flags: "u",
			options: {
				rewriteFlags: {
					unicode: false,
				}
			},
		},
		{
			options: {
				unicodePropertyEscape: false,
				rewriteFlags: {
					unicode: false,
				}
			},
		},
	] as {
		options?: IOptionsRewritePattern,
		flags?: string,
		expected?: {
			options?: IOptionsRewritePattern,
			flags?: string,
		}
	}[])
		.forEach(({ options, flags, expected }) =>
		{
			test(inspect({ options, flags }), () =>
			{

				let actual = handleOptions(options, flags);

				if (expected)
				{
					expect(actual).toStrictEqual(expected);
				}

				expect(actual.flags).toContain(EnumFlagMap.unicode);
				expect(actual.options).toHaveProperty('unicodePropertyEscape', true);

				expect(actual).toMatchSnapshot();
			})

		})

});

