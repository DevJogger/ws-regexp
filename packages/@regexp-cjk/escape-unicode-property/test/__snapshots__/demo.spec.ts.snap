// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allow any property like '\\\\P{UnicodeBinaryPropertyName}' 1`] = `
Array [
  "\\\\P{UnicodeBinaryPropertyName}",
  "P",
  "UnicodeBinaryPropertyName",
  undefined,
]
`;

exports[`allow any property like '\\\\p{UnicodePropertyName=UnicodePropertyValue}' 1`] = `
Array [
  "\\\\p{UnicodePropertyName=UnicodePropertyValue}",
  "p",
  "UnicodePropertyName",
  "UnicodePropertyValue",
]
`;

exports[`escapeUnicodePropertyPattern allow known property '\\\\P{Script_Extensions=Latin}' 1`] = `"[\\\\0-@\\\\[-\`\\\\{-\\\\xA9\\\\xAB-\\\\xB9\\\\xBB-\\\\xBF\\\\xD7\\\\xF7\\\\u02B9-\\\\u02DF\\\\u02E5-\\\\u0362\\\\u0370-\\\\u0484\\\\u0487-\\\\u0950\\\\u0953-\\\\u10FA\\\\u10FC-\\\\u1CFF\\\\u1D26-\\\\u1D2B\\\\u1D5D-\\\\u1D61\\\\u1D66-\\\\u1D6A\\\\u1D78\\\\u1DBF-\\\\u1DFF\\\\u1F00-\\\\u202E\\\\u2030-\\\\u2070\\\\u2072-\\\\u207E\\\\u2080-\\\\u208F\\\\u209D-\\\\u20EF\\\\u20F1-\\\\u2129\\\\u212C-\\\\u2131\\\\u2133-\\\\u214D\\\\u214F-\\\\u215F\\\\u2189-\\\\u2C5F\\\\u2C80-\\\\uA6FF\\\\uA708-\\\\uA721\\\\uA788-\\\\uA78A\\\\uA7CB-\\\\uA7CF\\\\uA7D2\\\\uA7D4\\\\uA7DA-\\\\uA7F1\\\\uA800-\\\\uA92D\\\\uA92F-\\\\uAB2F\\\\uAB5B\\\\uAB65\\\\uAB6A-\\\\uFAFF\\\\uFB07-\\\\uFF20\\\\uFF3B-\\\\uFF40\\\\uFF5B-\\\\u{1077F}\\\\u{10786}\\\\u{107B1}\\\\u{107BB}-\\\\u{1DEFF}\\\\u{1DF1F}-\\\\u{10FFFF}]"`;

exports[`escapeUnicodePropertyPattern allow known property '\\\\P{Script_Extensions=Latin}' 2`] = `
Array [
  "\\\\P{Script_Extensions=Latin}",
  "P",
  "Script_Extensions",
  "Latin",
]
`;

exports[`escapeUnicodePropertyPattern allow known property '\\\\p{Emoji_Presentation}' 1`] = `"[\\\\u231A\\\\u231B\\\\u23E9-\\\\u23EC\\\\u23F0\\\\u23F3\\\\u25FD\\\\u25FE\\\\u2614\\\\u2615\\\\u2648-\\\\u2653\\\\u267F\\\\u2693\\\\u26A1\\\\u26AA\\\\u26AB\\\\u26BD\\\\u26BE\\\\u26C4\\\\u26C5\\\\u26CE\\\\u26D4\\\\u26EA\\\\u26F2\\\\u26F3\\\\u26F5\\\\u26FA\\\\u26FD\\\\u2705\\\\u270A\\\\u270B\\\\u2728\\\\u274C\\\\u274E\\\\u2753-\\\\u2755\\\\u2757\\\\u2795-\\\\u2797\\\\u27B0\\\\u27BF\\\\u2B1B\\\\u2B1C\\\\u2B50\\\\u2B55\\\\u{1F004}\\\\u{1F0CF}\\\\u{1F18E}\\\\u{1F191}-\\\\u{1F19A}\\\\u{1F1E6}-\\\\u{1F1FF}\\\\u{1F201}\\\\u{1F21A}\\\\u{1F22F}\\\\u{1F232}-\\\\u{1F236}\\\\u{1F238}-\\\\u{1F23A}\\\\u{1F250}\\\\u{1F251}\\\\u{1F300}-\\\\u{1F320}\\\\u{1F32D}-\\\\u{1F335}\\\\u{1F337}-\\\\u{1F37C}\\\\u{1F37E}-\\\\u{1F393}\\\\u{1F3A0}-\\\\u{1F3CA}\\\\u{1F3CF}-\\\\u{1F3D3}\\\\u{1F3E0}-\\\\u{1F3F0}\\\\u{1F3F4}\\\\u{1F3F8}-\\\\u{1F43E}\\\\u{1F440}\\\\u{1F442}-\\\\u{1F4FC}\\\\u{1F4FF}-\\\\u{1F53D}\\\\u{1F54B}-\\\\u{1F54E}\\\\u{1F550}-\\\\u{1F567}\\\\u{1F57A}\\\\u{1F595}\\\\u{1F596}\\\\u{1F5A4}\\\\u{1F5FB}-\\\\u{1F64F}\\\\u{1F680}-\\\\u{1F6C5}\\\\u{1F6CC}\\\\u{1F6D0}-\\\\u{1F6D2}\\\\u{1F6D5}-\\\\u{1F6D7}\\\\u{1F6DD}-\\\\u{1F6DF}\\\\u{1F6EB}\\\\u{1F6EC}\\\\u{1F6F4}-\\\\u{1F6FC}\\\\u{1F7E0}-\\\\u{1F7EB}\\\\u{1F7F0}\\\\u{1F90C}-\\\\u{1F93A}\\\\u{1F93C}-\\\\u{1F945}\\\\u{1F947}-\\\\u{1F9FF}\\\\u{1FA70}-\\\\u{1FA74}\\\\u{1FA78}-\\\\u{1FA7C}\\\\u{1FA80}-\\\\u{1FA86}\\\\u{1FA90}-\\\\u{1FAAC}\\\\u{1FAB0}-\\\\u{1FABA}\\\\u{1FAC0}-\\\\u{1FAC5}\\\\u{1FAD0}-\\\\u{1FAD9}\\\\u{1FAE0}-\\\\u{1FAE7}\\\\u{1FAF0}-\\\\u{1FAF6}]"`;

exports[`escapeUnicodePropertyPattern allow known property '\\\\p{Emoji_Presentation}' 2`] = `
Array [
  "\\\\p{Emoji_Presentation}",
  "p",
  "Emoji_Presentation",
  undefined,
]
`;

exports[`handleOptions {
  options: { flags: 'usgi', rewriteFlags: { unicode: true } },
  flags: 'us'
} 1`] = `
Object {
  "flags": "usgi",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions {
  options: { unicodePropertyEscape: false, rewriteFlags: { unicode: false } },
  flags: undefined
} 1`] = `
Object {
  "flags": "u",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: { unicode: false } }, flags: 'u' } 1`] = `
Object {
  "flags": "u",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: { unicode: false } }, flags: 'us' } 1`] = `
Object {
  "flags": "us",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: { unicode: false } }, flags: undefined } 1`] = `
Object {
  "flags": "u",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: { unicode: true } }, flags: 'us' } 1`] = `
Object {
  "flags": "us",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: { rewriteFlags: {} }, flags: 'us' } 1`] = `
Object {
  "flags": "us",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: undefined, flags: '' } 1`] = `
Object {
  "flags": "u",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`handleOptions { options: undefined, flags: undefined } 1`] = `
Object {
  "flags": "u",
  "options": Object {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "rewriteFlags": Object {
      "unicode": true,
    },
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;
